packages/api/src/clients/index.ts

```diff
@@ -44,8 +46,10 @@ export class Client {
       });
     }
 
-    this.bots = new BotClient(this.http);
-    this.users = new UserClient(this.http);
+    this._clientSettings = {...DEFAULT_CLIENT_SETTINGS, ...(clientSettings ?? {})};
```

Comment:
this is copied in many places. Consider a helper method that takes in `Partial<ClientSettings>` and returns `ClientSettings`